repos/system_upgrade/el8toel9/actors/checkifcfg/libraries/ifcfg.py

```diff
@@ -0,0 +1,143 @@
+            continue
+
+        if f == 'ifcfg-lo':
+            continue
+
+        for line in open(path).readlines():
+            try:
+                (key, value) = line.split('#')[0].strip().split('=')
+            except ValueError:
+                # We're not interested in lines that are not
+                # simple assignments. Play it safe.
+                continue
+
+            if key in ('TYPE', 'DEVICETYPE'):
+                if got_type is None:
+                    got_type = value.lower()
+                elif got_type != value.lower():
+                    bad_type = True
+
+            if key == 'BONDING_MASTER':
+                if got_type is None:
+                    got_type = 'bond'
+                elif got_type != 'bond':
+                    bad_type = True
+
+            if key == 'NM_CONTROLLED' and value.lower() not in TRUE_VALUES:
+                nm_controlled = False
+
+        if got_type in TYPE_MAP:
+            if TYPE_MAP[got_type] not in rpms_to_install:
+                rpms_to_install.append(TYPE_MAP[got_type])
+        else:
+            bad_type = True
+
+        # Don't bother reporting the file for NM_CONTROLLED=no
+        # if its type is not supportable with NetworkManager anyway
+        if bad_type is True:
+            bad_type_files.append(path)
+        elif nm_controlled is False:
+            not_controlled_files.append(path)
+
+    if bad_type_files:
+        title = 'Network configuration for unsupported device types detected'
+        summary = ('RHEL 9 does not support the legacy network-scripts'
+                   ' package that was deprecated in RHEL 8 in favor of'
+                   ' NetworkManager. Files for device types that are not'
+                   ' supported by NetworkManager are present in the system.')
+        remediation = ('Consult the nm-settings-ifcfg-rh(5) manual for'
+                       ' valid types of ifcfg files. Remove configuration'
+                       ' files that can not be supported.')
+        reporting.create_report([
+            reporting.Title(title),
+            reporting.Summary(summary),
+            reporting.Remediation(hint=remediation),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Tags([reporting.Tags.NETWORK, reporting.Tags.SERVICES]),
+            reporting.Flags([reporting.Flags.INHIBITOR]),
+            reporting.RelatedResource('package', 'NetworkManager'),
+        ] + [
+            reporting.RelatedResource('file', fname)
+            for fname in bad_type_files
+        ])
+
+    if not_controlled_files:
+        title = 'Network configuration with disabled NetworkManager support detected'
+        summary = ('RHEL 9 does not support the legacy network-scripts'
+                   ' package that was deprecated in RHEL 8 in favor of'
+                   ' NetworkManager. Configuration present in the system'
+                   ' prohibit NetworkManager from loading it.')
```

Comment:
@lkundrak I suggest to add the list of bad files into the summary as well. This report is good for users reading it in cockpit or satellite, but in case people read just the txt form, addititional fields like "resources" are not printed to user.

```suggestion
        FMT_LIST_SEPARATOR = '\n    - '
        summary = ('RHEL 9 does not support the legacy network-scripts'
                   ' package that was deprecated in RHEL 8 in favor of'
                   ' NetworkManager. Configuration present in the system'
                   ' prohibit NetworkManager from loading it. Files with the problematic configuration:{}'
        ).format(''.join(['{}{}'.format(FMT_LIST_SEPARATOR, bfile) for bfile in bad_files]))
```